### DISCUSSION TITLE
create new user programmatically

### DISCUSSION THREAD

**Discussion by JustinGuese** [fief-dev/fief#70] [Q&A] [answered]
could someone point me in the right direction on how to create a user programmatically?
like if I created the "first" admin user in the hosted fief GUI, how could I write a route that can create a new user and assign that user a role?

thanks for your help!

**Comment by frankie567** [comment, marked as answer]
Hi @JustinGuese, welcome to Fief 👋 

The best way is to use the Admin API for this. Documentation is a bit missing for this but basically you need two things:

* An [Admin API key](https://docs.fief.dev/getting-started/api-keys/).
* Make HTTP requests to the API. There is an interactive documentation available at your workspace subdomain. For example, if your subdomain is `justinguese`, you'll find it at `https://justinguese.fief.dev/admin/api/docs`.
    * To create a user, you'll be interested in `POST /users/`.
    * To assign a role, you'll be interested in `POST /users/{id}/roles`.
    * The Admin API Key should be passed in the `Authorization` header with the `Bearer` scheme: `Authorization: Bearer YOUR_KEY`.

Of course, users can also create an account themselves on the authentication pages 😉
